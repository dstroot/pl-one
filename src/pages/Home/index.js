import React, { useEffect } from 'react';
import './styles.scss';

import HeroWash from '../../components/HeroWash';
import logo from '../../components/HeroWash/media/PacificLife_BrandIcon_White.png';
import TextHero from '../../components/TextHero';
import CardGroup from '../../components/CardGroup';
import Search from '../../components/Search';
import Chart from '../../components/Chart';

const Home = () => {
  // Set the page title and position using the useEffect hook
  useEffect(() => {
    document.title = `Pacific Life • Home`;
    window.scrollTo(0, 0);
  });

  return (
    <>
      <HeroWash
        logo={logo}
        bgimage="https://www.pacificlife.com/content/dam/paclife/crp/images/brand/Brand_hero.jpg"
        tagline="You. Protected."
        line1="Our mission is protecting your financial future."
        line2="We’ve fulfilled that promise for 150 years."
      />
      <TextHero
        heading="PROTECTING WHAT MATTERS TO YOU MOST FOR MORE THAN 150 YEARS"
        text="When it comes to life, no two people are the same. You're unique and so are your financial goals. For more than 150 years, generations of families and businesses have harnessed the Power of Pacific to help them achieve their unique goals. Explore how Pacific Life can help you."
        link="https://www.pacificlife.com/home/life-goals.html"
      />
      <CardGroup />
      <Chart />
      <Search />
    </>
  );
};

export default Home;
