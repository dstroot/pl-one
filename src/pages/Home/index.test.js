import React from 'react';
import { create } from 'react-test-renderer';
import Home from '../Home';

function createNodeMock() {
  const doc = document.implementation.createHTMLDocument();
  return { parentElement: doc.body };
}

describe('Home', () => {
  it('it should render', () => {
    let tree = create(<Home />, { createNodeMock });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
