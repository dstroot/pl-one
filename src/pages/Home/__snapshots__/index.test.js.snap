// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home it should render 1`] = `
Array [
  <div
    style={
      Object {
        "backgroundImage": "linear-gradient(to top, rgba(0, 142, 215, 0.9), rgba(0, 142, 215, 0.4)), url(https://www.pacificlife.com/content/dam/paclife/crp/images/brand/Brand_hero.jpg)",
        "backgroundPosition": "center",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "cover",
        "height": "40vw",
        "minHeight": "325px",
        "position": "relative",
      }
    }
  >
    <div
      style={
        Object {
          "bottom": "5vw",
          "color": "white",
          "position": "absolute",
          "textAlign": "center",
          "width": "100vw",
        }
      }
    >
      <img
        alt="hero"
        className="d-block-inline"
        src="PacificLife_BrandIcon_White.png"
      />
      <p
        style={
          Object {
            "fontSize": "calc(32px + 1.5vw + 0.5vmin)",
            "fontWeight": "500",
            "marginBottom": "0",
          }
        }
      >
        You. Protected.
      </p>
      <p
        style={
          Object {
            "fontSize": "calc(8px + 1.5vw + 0.5vmin)",
            "fontWeight": "300",
            "marginBottom": "0",
            "marginTop": "-0.5vw",
          }
        }
      >
        Our mission is protecting your financial future.
      </p>
      <p
        style={
          Object {
            "fontSize": "calc(8px + 1.5vw + 0.5vmin)",
            "fontWeight": "300",
            "marginBottom": "0",
            "marginTop": "-0.5vw",
          }
        }
      >
        Weâ€™ve fulfilled that promise for 150 years.
      </p>
    </div>
  </div>,
  <div
    className="container mt-5 mb-5"
  >
    <div
      className="row justify-content-center"
    >
      <div
        className="col"
      >
        <div
          className="text-center mx-auto"
        >
          <p
            style={
              Object {
                "fontSize": "calc(26px + 1.5vw + 0.5vmin)",
                "fontWeight": "300",
                "lineHeight": "1.2",
              }
            }
          >
            PROTECTING WHAT MATTERS TO YOU MOST FOR MORE THAN 150 YEARS
          </p>
          <p
            className="lead"
          >
            When it comes to life, no two people are the same. You're unique and so are your financial goals. For more than 150 years, generations of families and businesses have harnessed the Power of Pacific to help them achieve their unique goals. Explore how Pacific Life can help you.
          </p>
          <a
            className="btn btn-primary"
            href="https://www.pacificlife.com/home/life-goals.html"
            target="_self"
          >
            We can help!
          </a>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="container-fluid bg-primary text-center text-white"
  >
    <div
      className="row"
    >
      <div
        className="col-10 offset-1"
      >
        <p
          className="pt-5"
          style={
            Object {
              "fontSize": "calc(26px + 1.5vw + 0.5vmin)",
              "fontWeight": "300",
              "lineHeight": "1.2",
            }
          }
        >
          What the Power of Pacific Means for You
        </p>
        <div
          className="card-group my-3"
        >
          <div
            className="card bg-primary border-0"
          >
            <div
              className="svg-container mx-auto"
            >
              <svg>
                flag.svg
              </svg>
            </div>
            <div
              className="card-body"
            >
              <h3
                className="card-title"
              >
                Freedom
              </h3>
              <p
                className="card-text"
              >
                Life insurance and retirement solutions to help you reach your goals.
              </p>
            </div>
          </div>
          <div
            className="card bg-primary border-0"
          >
            <div
              className="svg-container mx-auto"
            >
              <svg>
                locked.svg
              </svg>
            </div>
            <div
              className="card-body"
            >
              <h3
                className="card-title"
              >
                Security
              </h3>
              <p
                className="card-text"
              >
                A deep commitment to protecting generations of families.
              </p>
            </div>
          </div>
          <div
            className="card bg-primary border-0"
          >
            <div
              className="svg-container d-block mx-auto align-bottom"
            >
              <svg>
                handshake.svg
              </svg>
            </div>
            <div
              className="card-body"
            >
              <h3
                className="card-title"
              >
                Trust
              </h3>
              <p
                className="card-text"
              >
                150 years of strength and stability by your side.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="container mt-5 mb-4"
  >
    <div
      className="row"
    >
      <div
        className="col-md-6"
      >
        <svg
          className="d-block w-100"
          height={300}
        >
          saving3.svg
        </svg>
      </div>
      <div
        className="col-md-6 text-center mt-3"
      >
        <h1
          style={
            Object {
              "fontSize": "calc(36px + 1vw)",
              "fontWeight": "300",
              "marginBottom": "0",
              "marginTop": "30",
            }
          }
        >
          Design Your Future
        </h1>
        <p
          className="lead text-muted mt-0"
        >
          See how fast your money can grow!
        </p>
        <div
          className="pl-4 pr-4"
        >
          <div
            className="mt-3"
          >
            <div
              className="d-flex justify-content-between"
            >
              <div
                className="text-muted"
              >
                Initial Deposit
              </div>
              <div>
                <h4
                  className="text-primary"
                >
                  $50,000
                </h4>
              </div>
            </div>
            <div
              className="rc-slider"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseUp={[Function]}
              onTouchStart={[Function]}
            >
              <div
                className="rc-slider-rail"
                style={Object {}}
              />
              <div
                className="rc-slider-track"
                style={
                  Object {
                    "left": "0%",
                    "width": "60.60606060606061%",
                  }
                }
              />
              <div
                className="rc-slider-step"
              />
              <div
                aria-disabled={false}
                aria-valuemax={99}
                aria-valuemin={0}
                aria-valuenow={60}
                className="rc-slider-handle"
                onBlur={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="slider"
                style={
                  Object {
                    "left": "60.60606060606061%",
                  }
                }
                tabIndex={0}
              />
              <div
                className="rc-slider-mark"
              />
            </div>
          </div>
          <div
            className="mt-3"
          >
            <div
              className="d-flex justify-content-between"
            >
              <div
                className="text-muted"
              >
                Monthly Deposit
              </div>
              <div>
                <h4
                  className="text-primary"
                >
                  $1,000
                </h4>
              </div>
            </div>
            <div
              className="rc-slider"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseUp={[Function]}
              onTouchStart={[Function]}
            >
              <div
                className="rc-slider-rail"
                style={Object {}}
              />
              <div
                className="rc-slider-track"
                style={
                  Object {
                    "left": "0%",
                    "width": "53.535353535353536%",
                  }
                }
              />
              <div
                className="rc-slider-step"
              />
              <div
                aria-disabled={false}
                aria-valuemax={99}
                aria-valuemin={0}
                aria-valuenow={53}
                className="rc-slider-handle"
                onBlur={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="slider"
                style={
                  Object {
                    "left": "53.535353535353536%",
                  }
                }
                tabIndex={0}
              />
              <div
                className="rc-slider-mark"
              />
            </div>
          </div>
          <div
            className="mt-3"
          >
            <div
              className="d-flex justify-content-between"
            >
              <div
                className="text-muted"
              >
                Risk Level
              </div>
              <div>
                <h4
                  className="text-primary"
                >
                  Growth
                </h4>
              </div>
            </div>
          </div>
          <div
            className="rc-slider"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="rc-slider-rail"
              style={Object {}}
            />
            <div
              className="rc-slider-track"
              style={
                Object {
                  "left": "0%",
                  "width": "44.44444444444444%",
                }
              }
            />
            <div
              className="rc-slider-step"
            >
              <span
                className="rc-slider-dot rc-slider-dot-active"
                style={
                  Object {
                    "left": "0%",
                  }
                }
              />
              <span
                className="rc-slider-dot rc-slider-dot-active"
                style={
                  Object {
                    "left": "11.11111111111111%",
                  }
                }
              />
              <span
                className="rc-slider-dot rc-slider-dot-active"
                style={
                  Object {
                    "left": "22.22222222222222%",
                  }
                }
              />
              <span
                className="rc-slider-dot rc-slider-dot-active"
                style={
                  Object {
                    "left": "33.33333333333333%",
                  }
                }
              />
              <span
                className="rc-slider-dot rc-slider-dot-active"
                style={
                  Object {
                    "left": "44.44444444444444%",
                  }
                }
              />
              <span
                className="rc-slider-dot"
                style={
                  Object {
                    "left": "55.55555555555556%",
                  }
                }
              />
              <span
                className="rc-slider-dot"
                style={
                  Object {
                    "left": "66.66666666666666%",
                  }
                }
              />
              <span
                className="rc-slider-dot"
                style={
                  Object {
                    "left": "77.77777777777779%",
                  }
                }
              />
              <span
                className="rc-slider-dot"
                style={
                  Object {
                    "left": "88.88888888888889%",
                  }
                }
              />
              <span
                className="rc-slider-dot"
                style={
                  Object {
                    "left": "100%",
                  }
                }
              />
            </div>
            <div
              aria-disabled={false}
              aria-valuemax={9}
              aria-valuemin={0}
              aria-valuenow={4}
              className="rc-slider-handle"
              onBlur={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              role="slider"
              style={
                Object {
                  "left": "44.44444444444444%",
                }
              }
              tabIndex={0}
            />
            <div
              className="rc-slider-mark"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="row mt-3"
    >
      <div
        className="col-md-12 mt-3"
      >
        <div
          className="recharts-responsive-container"
          style={
            Object {
              "height": 400,
              "maxHeight": undefined,
              "minHeight": undefined,
              "minWidth": undefined,
              "width": "100%",
            }
          }
        >
          <div
            style={
              Object {
                "display": "none",
                "height": 0,
                "position": "absolute",
                "visibility": "hidden",
                "width": 0,
              }
            }
          />
        </div>
        <div
          className="d-flex justify-content-between mb-3 text-muted"
        >
          <div
            className="p-2 font-weight-bold ml-5"
          >
            Today
          </div>
          <div
            className="p-2 font-weight-bold"
          >
            30 Years
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="container mt-5"
  >
    <div
      className="row align-items-center"
    >
      <div
        className="col-md-6"
      >
        <h1
          style={
            Object {
              "fontSize": "calc(24px + 1vw)",
              "fontWeight": "300",
              "marginBottom": "0",
            }
          }
        >
          CONFIDENCE IN YOUR FUTURE STARTS HERE
        </h1>
        <h3
          className=""
        >
          Find a Financial Professional
        </h3>
        <form>
          <div
            className="form-row align-items-center"
          >
            <div
              className="col-auto"
            >
              <input
                aria-label="zip code"
                className="form-control mb-2"
                id="zip"
                placeholder="enter zip code"
                type="text"
              />
            </div>
            <div
              className="col-auto"
            >
              <button
                className="btn btn-primary mb-2"
                type="submit"
              >
                Submit
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
    <div
      className="row mt-3"
    >
      <div
        className="col"
      >
        <p
          className="text-muted small"
        >
          In order to sell life insurance, a financial professional must be a properly licensed and appointed life insurance producer.
            
          <span
            className=""
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "fontWeight": "600",
              }
            }
          >
            [ 
            More +
             ]
          </span>
        </p>
      </div>
    </div>
  </div>,
]
`;
